### PR TITLE
app-crypt/tpm2-tss: Don't fail tmpfiles_process where /sys is restricted

### PR DESCRIPTION
**Pull request by salahcoronya**
Closes: https://bugs.gentoo.org/893636

### CODE REVIEW

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+DESCRIPTION="TCG Trusted Platform Module 2.0 Software Stack"
+HOMEPAGE="https://github.com/tpm2-software/tpm2-tss"
+SRC_URI="https://github.com/tpm2-software/${PN}/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0/4"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+IUSE="doc +fapi +openssl mbedtls +policy static-libs test"
+
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:16]: This empty line looks unnecessary.

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="^^ ( mbedtls openssl )
+		fapi? ( openssl !mbedtls )
+		policy? ( openssl !mbedtls )"
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:21]:
```suggestion
REQUIRED_USE="
	^^ ( mbedtls openssl )
	fapi? ( openssl !mbedtls )
	policy? ( openssl !mbedtls )
"
```

Improves readability, makes changes easier and keeps indent consistent.

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+RDEPEND="acct-group/tss
+	acct-user/tss
+	sys-apps/util-linux:=[${MULTILIB_USEDEP}]
+	fapi? ( dev-libs/json-c:=[${MULTILIB_USEDEP}]
+		>=net-misc/curl-7.80.0[${MULTILIB_USEDEP}] )
+	mbedtls? ( net-libs/mbedtls:=[${MULTILIB_USEDEP}] )
+	openssl? ( dev-libs/openssl:=[${MULTILIB_USEDEP}] )"
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:29]:
```suggestion
RDEPEND="
	acct-group/tss
	acct-user/tss
	sys-apps/util-linux:=[${MULTILIB_USEDEP}]
	fapi? (
		dev-libs/json-c:=[${MULTILIB_USEDEP}]
		>=net-misc/curl-7.80.0[${MULTILIB_USEDEP}]
	)
	mbedtls? ( net-libs/mbedtls:=[${MULTILIB_USEDEP}] )
	openssl? ( dev-libs/openssl:=[${MULTILIB_USEDEP}] )
"
```

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+DEPEND="${RDEPEND}
+	test? ( app-crypt/swtpm
+		dev-libs/uthash
+		dev-util/cmocka
+		fapi? ( >=net-misc/curl-7.80.0 ) )"
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:35]:
```suggestion
DEPEND="
	${RDEPEND}
	test? (
		app-crypt/swtpm
		dev-libs/uthash
		dev-util/cmocka
		fapi? ( >=net-misc/curl-7.80.0 )
	)
"
```

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+BDEPEND="sys-apps/acl
+	virtual/pkgconfig
+	doc? ( app-text/doxygen )"
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:38]:
```suggestion
BDEPEND="
	sys-apps/acl
	virtual/pkgconfig
	doc? ( app-text/doxygen )
"
```

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${PN}-4.0.0-Dont-install-files-into-run.patch"
+	"${FILESDIR}/${PN}-4.0.1-Make-sysusers-and-tmpfiles-optional.patch"
+	"${FILESDIR}/${PN}-4.0.1-Do-not-consider-failures-to-write-files-in-sys-hard.patch"
+	)
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:44]:
```suggestion
)
```

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+pkg_setup() {
+	local CONFIG_CHECK=" \
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:47]: Do you actually need that backslash for anything? Alternatively, you could make it all one line.

File: app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+		$(multilib_native_use_enable test self-generated-certificate) \
+		--disable-tcti-libtpms \
+		--disable-defaultflags \
+		--disable-weakcrypto \
+		--with-crypto="$(usex mbedtls mbed ossl)" \
+		--with-runstatedir=/run \
+		--with-udevrulesdir="$(get_udevdir)/rules.d" \
+		--with-udevrulesprefix=60- \
+		--without-sysusersdir \
+		--with-tmpfilesdir="/usr/lib/tmpfiles.d"
```
Comment (mgorny) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.1-r1.ebuild:82]:
```suggestion
	local myconf=(
		--localstatedir=/var
		$(multilib_native_use_enable doc doxygen-doc)
		$(use_enable fapi)
		$(use_enable policy)
		$(use_enable static-libs static)
		$(multilib_native_use_enable test unit)
		$(multilib_native_use_enable test integration)
		$(multilib_native_use_enable test self-generated-certificate)
		--disable-tcti-libtpms
		--disable-defaultflags
		--disable-weakcrypto
		--with-crypto="$(usex mbedtls mbed ossl)"
		--with-runstatedir=/run
		--with-udevrulesdir="$(get_udevdir)/rules.d"
		--with-udevrulesprefix=60-
		--without-sysusersdir
		--with-tmpfilesdir="/usr/lib/tmpfiles.d"
	)
	ECONF_SOURCE=${S} econf "${myconf[@]}"
```

Avoiding backslashes makes maintenance easier and avoids potential mistakes when editing.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by salahcoronya** [comment]
I made all the specified changes

**Comment by mgorny** [comment]
Thanks!

**Pull request closed without merging**